Lesson 12

Hanif947 opened the pull request.

Hanif947 committed:
Feat after 6th lesson:
1. Added a function, linked it all to a template
 so that the text "About the company" is taken from
 the aboutCompany class field;
2. Added a button that will hide/show the main
 large picture banner;
3. I made the numbers 1 2 3 4 5 to be replaced
 by 5 4 3 2 1 using a loop through the array
 via *ngFor;
4. Added a "Change text" button, when clicked,
 the menu text will become small/large.

Hanif947 committed:
Feat after 7th lesson:
1. Moved the site header into a separate component
   and connected it to app.component;
2. Improved the cards a bit, added shadows,
   decorated the button;
3. Made a redirect to the /users page when clicking
   on the button in the site header
   (new button “Страница users”);
4. Added interface.ts file and added
   User[] interface to it

Hanif947 committed:
Feat:
Added users-api.service; added user.interface.ts;
added user-card.component;
added users-list.component to routing;
added "delete user" button to user-card;
added todo page, and added all of the above to todo.

Fix:
changed the name of user.interface.ts,
and changed the interface User[],
removed the typing stub - any.

Hanif947 committed:
Feat: Rewrote users and todos to RxJS.
Fix: Corrected formatting.

Hanif947 committed:
Feat: Angular forms for creating users and todos

Hanif947 committed:
Feat: connected the angular material library
to the project and used it to style the form for
creating users and todos

pocahontassss commented on src/app/todos-list/todos-list.component.ts:
```diff
@@ -4,16 +4,27 @@ import { Todo } from '../todo.interface';
 })
 export class TodosListComponent {
+
+   public createTodo(formData: any) {
```
тип

pocahontassss commented on src/app/create-user-form/create-user-form.component.ts:
```diff
@@ -0,0 +1,29 @@
+export class CreateUserFormComponent {
+
+  @Output()
+  createUser = new EventEmitter();
```
тип

pocahontassss commented on src/app/create-todo-form/create-todo-form.component.ts:
```diff
@@ -0,0 +1,53 @@
+export class CreateTodoFormComponent {
+
+  @Output()
+  createTodo = new EventEmitter();
```
тип

pocahontassss commented on src/app/create-todo-form/create-todo-form.component.ts:
```diff
@@ -0,0 +1,53 @@
+    completed: new FormControl('', [Validators.required, completedValidator()])
+  })
+
+    private getCompletedValue(): boolean {
+    const value = this.form.get('completed')?.value!.trim().toLowerCase();
+    if (value === 'yes') 
+      return true;
+    else return false;
+  }
```
Форматирование

pocahontassss commented on src/app/footer/footer.component.scss:
```diff
@@ -0,0 +1,104 @@
+                border-radius: 5px;
+                cursor: pointer;
+                font-weight: bold;
+            }
```
Не используем стилизацию по тегам. Не ленитесь 1 раз класс написать, но не нужно писать теги

pocahontassss commented on src/app/users.service.ts:
```diff
@@ -0,0 +1,45 @@
+
+    createUser(user: User) { 
+        const existingUser = this.usersSubject$.value.find(
+            currentElement => currentElement.email === user.email
```
тип

pocahontassss commented on src/app/header/header.component.ts:
```diff
@@ -0,0 +1,39 @@
+
+  changeMenuText() {
+    this.menuItems = upperCaseMenuItems.map(
+      item => this.isUpperCase ? item.toLowerCase() : item.toUpperCase()
```
тип

Hanif947 committed:
Feat:
connected the Angular Material library to
the project and used it to style the form for
creating users and todos;

Feat: connected modal dialogs and snackbar to users and todos;

Fix: fixed some issues in the previous code
